src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -224,36 +225,48 @@ public async Task<Diagnostic> ToDiagnosticAsync(Project project, CancellationTok
                 location, additionalLocations, customTags: CustomTags, properties: Properties);
         }
 
-        public static TextSpan GetTextSpan(DiagnosticDataLocation? dataLocation, SourceText text)
+        public static (LinePosition startLine, LinePosition endLine) GetLinePositions(DiagnosticDataLocation? dataLocation, SourceText text, bool useMapped)
         {
             var lines = text.Lines;
             if (lines.Count == 0)
             {
                 return default;
             }
 
-            var originalStartLine = dataLocation?.OriginalStartLine ?? 0;
-            if (originalStartLine >= lines.Count)
+            var dataLocationStartLine = (useMapped ? dataLocation?.MappedStartLine : dataLocation?.OriginalStartLine) ?? 0;
+            var dataLocationStartColumn = (useMapped ? dataLocation?.MappedStartColumn : dataLocation?.OriginalStartColumn) ?? 0;
+            var dataLocationEndLine = (useMapped ? dataLocation?.MappedEndLine : dataLocation?.OriginalEndLine) ?? 0;
+            var dataLocationEndColumn = (useMapped ? dataLocation?.MappedEndColumn : dataLocation?.OriginalEndColumn) ?? 0;
```

Comment:
when is useMapped false?

Reply:
The tagger doesn't use the mapped spans - http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Diagnostics/DiagnosticData.cs,828891a1bfd06025

http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.cs,240

I didn't want to change that so added the flag.